decidim-core/lib/decidim/filter_form_builder.rb

```diff
@@ -33,38 +33,56 @@ def collection_check_boxes(method, collection, value_method, label_method, optio
       end
     end
 
+    # Wrap the dependant check_boxes in a custom fieldset.
+    # checked parent checks its children
+    def chained_check_boxes(method, collection, options = {})
```

Comment:
Would you consider renaming to `check_boxes_tree`?

Reply:
I've commited this change, let's check that everything keeps working and it should be ready. Regarding the [product request](https://github.com/decidim/decidim/issues/5602#issuecomment-585091993), it is already commited [here](https://github.com/decidim/decidim/pull/5654/commits/da0b92ca548bcd50fb1866f0d784557439635181).